### DISCUSSION TITLE
Unable to find a way to directly tokenize a Credit card

### DISCUSSION THREAD

**Discussion by nikhil-varma** [swellstores/community#516] [Help] [answered]
I am using `swell-node` to work with my app. What I am trying to do is create a card for a customer account via the Backend API however, I am unsure what URL or API method to call to get this done. I went through this documentation [Direct card tokenization](https://developers.swell.is/frontend-api/payments#create-a-card-token) but somehow that doesn't seem to work. I get a `Cannot read createToken of undefined` TS error despite initialising the Swell's object via `swell.init` 

Can anyone help me understand what am I missing or is there a completely different approach to this? I see that there is a FrontendAPI but I am not looking at using that. I need an approach via the BackendAPIs and would greatly appreciate any help around this.

**Comment by nikhil-varma** [comment]
I have also checked the GraphQL part of it but the documentation doesn't tell anything specific via GraphQL either. It is pointing it out to the `swell.card.createToken` call which from the page linked above.

**Comment by Ojay** [comment]
I've also come across the above issues... seems like we have to use `swell-js` just for this one task of tokenisation.  I would also love some clarity on a best practice approach to tokenisation with either `swell-node` or via the GraphQl endpoint. Thanks so much!

**Comment by Ojay** [comment]
Just to elaborate a bit more on this.

I spent a lot of time setting up Swell in my Nextjs application using your examples as a guide. This [repo](https://github.com/swellstores/swell-next-graphql-example) in conjunction with your [blog post guide](https://www.swell.is/blog/swell-next-js-graphql-building-modern-ecommerce-experiences).

The [front-end documentation ](https://developers.swell.is/frontend-api/introduction) is presented as either or. Cool, GraphQl it is. That is until you get down the road and realise, at least this seems to be the case, that things like tokenisation can only be achieved with the Swell.js library. So now it looks like I'm going to have to use the GraphQl api, `swell-node` & `swell-js` to achieve a fully functioning integration.

This seems a little crazy, does anyone care to advise on the tokenisation scenario with the various libraries? Seems the GraphQl endpoint is only really suitable for basic setups?

Any insight into this would be very welcome. Thanks!

**Reply by sags95** [reply]
Payments is one area that is lacking with the current GQL implementation. We will be publishing some additional context in the docs site to provide better clarity around these limitations shortly. For now, tokenization is only supported with `swell-js`, either through direct card tokenization, or with the `tokenize()` when using Stripe Elements. `swell-node` does not support creating card tokens at this time.

**Reply by Ojay** [reply]
I appreciate the clarity but that's a lot of work that needs scrapping. My bad for not researching the library properly, but also, your bad for not communicating the limitations with the libraries either.

Ahh well, at least we know now. For what its worth everything else has been a joy, it's just a real shame to have come this far and have to now rethink and rework a few things.

**Reply by nikhil-varma** [reply]
I agree with @Ojay on this. 
It is also very difficult to find what is required from the docs. Please consider improving that as well.

**Reply by ericingram** [reply, marked as answer]
We made a lot of mistakes with the GraphQL implementation. As @sags95 mentioned, we're about to add some warnings to the docs to clarify those limitations. The truth is that the way it was first implemented won't be able to support the fully flexibility of the platform and it needs to be entirely revamped for that reason. There's no ETA for this effort either as we have other priorities currently.

I would generally advise against using GQL until we get there. Really sorry for the confusion and trouble.
